test/inductor/test_mkldnn_pattern_matcher.py

```diff
@@ -3632,18 +3661,28 @@ def __init__(self, dtype: torch.dtype, has_bias: bool):
                 self.b_scale = self.b_scale.to(dtype)
                 self.bias = torch.rand([out_feature], dtype=dtype) if has_bias else None
 
+                self.additive = torch.rand([M, out_feature], dtype=dtype)
+
             def forward(self, a):
                 if reshape_a:
                     a_reshaped = a.reshape(-1, a.size(-1))
                 else:
                     a_reshaped = a
                 c = torch._int_mm(a_reshaped, self.b)
                 c = c.to(self.dtype)
-                a_scale = self.a_scale.expand(c.shape)
+                if expand_a_scale:
+                    a_scale = self.a_scale.expand(c.shape)
+                else:
+                    a_scale = self.a_scale
                 c = c * a_scale
                 c = c * self.b_scale
                 if self.has_bias:
                     c = c + self.bias
+                elif inplace_add and test_for_pointwise_binary:
+                    # When M is 1, dynamic shapes are enabled with torch.compile, has_bias is False,
```

Comment:
When M is 1, I think Inductor will always do the specialization instead of using the symbolic size.

Reply:
Tracking that issue at #148199. Thanks!